ibm-openbmc/phosphor-certificate-manager
# Added methods for IBM-ACF upload functionality.

These methods allow the upload and info
    getting for ACF files.
    Method names respectively are installACF and getACFInfo

Supported functions were also added for validating the ACF
    on upload. Now links with ibm-acf for functionality.

installACF is meant to validate the ACF file contents passed
    as the argument.
    If valid then install file. Do not install file if invalid.
    If ACFfile contents are empty, treat as request to delete
        installed acf.
    Returns the contents of the installed ACF

getACFInfo validates installed ACF file and returns ACF
    contents (accessControlFile) and ACF info
    (isAcfInstalled == false, sDate == "")

This ACF interface is intended to be called from bmcweb.
    bmcweb receives an encoded ACF file,decodes it,
    calls this intf.method through dbus, validates
    contents, and returns info back to the request

    If getACFInfo method it isn't passed an argument
    and validates the currently install ACF file and
    returns info about the installed file

Tested:
    Test case 1
    1) Ensure ACF is not present
    2) Call getACFInfo which returns:
        (accessControlFile == {},
        isAcfInstalled == false,
        sDate == "")

    Test case 2
    1) Ensure ACF is present
    2) Call getACFInfo which returns:
        (accessControlFile == {binaryData},
        isAcfInstalled == true,
        sDate == "ACF expiration")

    Test case 3
    1) Ensure ACF is not present
    2) Call installACF with ACFFile binary contents
    3) Valid ACF passes validation:
        (Signature, serial number, expiration)
    4) File contents are written to install destination
    5) Call returns:
        (accessControlFile == {binaryData},
        isAcfInstalled == true,
        sDate == "ACF expiration")

    Test case 4
    1) Ensure ACF is not present
    2) Call installACF with ACFFile binary contents
    3) Invalid ACF fails validation:
        (Signature, serial number, expiration)
    4) ACF is not installed
    5) Call returns:
        (accessControlFile == {},
        isAcfInstalled == false,
        sDate == "")
